### ISSUE TITLE
New version is wrong and cause a fail of pipeline

### ISSUE THREAD

**Issue by MaximStone** [undergroundwires/bump-everywhere#13]
![image](https://user-images.githubusercontent.com/9432158/146018614-75727163-f413-4e38-840e-06f6ab76e067.png)

Version of bump-everywhere 1.3.0
But as I remember it started before I updated version.

I had reading the shell scripts for several days, but no luck.

```
Cloning Intentop/love-forever
Cloning into '/tmp/tmp.KjNkCL'...
Latest commit sha: 251a4907968843ecfa0efeeffaf5ec0b86c88859
Checking if this run is a re-run
Setting up credentials
Bumping and tagging version
Updating "1.5.3" to "1 5 3..1"
Creating tag: "1 5 3..1"
fatal: '1 5 3..1' is not a valid tag name.
Could not tag: "1 5 3..1"
Could not bump & tag
```

I suppose it's on my side but where? Wrong dots? ;)

**Comment by undergroundwires** [comment]
Hi @MaximStone , thanks for the bug report.

The problem is [`increase_patch_version`](https://github.com/undergroundwires/bump-everywhere/blob/1adca90930f04a5486f30b0758cd62265464f510/scripts/bump-and-tag-version.sh#L33-L41). Specifically this line `version_bits=("${version//./ }")` went bananas with double quotes, no double quotes should be there instead be `version_bits=(${version//./ }"`.

I changed the workflow I use this action. Instead of letting it patch, I run it when I push tags (patch version myself), it then skips bumping the version but does everything else, so I did not realize this bug in new version. I will prepare an elegant fix as soon with proper testing.

Happy to know that someone else uses this action too, I thought I was the only one 😄 Thanks for the report!

**Comment by undergroundwires** [comment]
The problem is [`increase_patch_version`](https://github.com/undergroundwires/bump-everywhere/blob/1adca90930f04a5486f30b0758cd62265464f510/scripts/bump-and-tag-version.sh#L33-L41). Specifically this line `version_bits=("${version//./ }")` went bananas with double quotes, no double quotes should be there instead be `version_bits=(${version//./ }"`.

I changed the workflow I use this action. Instead of letting it patch, I run it when I push tags (patch version myself), it then skips bumping the version but does everything else, so I did not realize this bug in new version. I will prepare an elegant fix as soon with proper testing.

Happy to know that someone else uses this action too, I thought I was the only one 😄 Thanks for the report!

**Comment by undergroundwires** [comment]
The problem is [`increase_patch_version`](https://github.com/undergroundwires/bump-everywhere/blob/1adca90930f04a5486f30b0758cd62265464f510/scripts/bump-and-tag-version.sh#L33-L41). Specifically this line `version_bits=("${version//./ }")` went bananas with double quotes since aff2c239f356e6a757b8efdb14cf0526f421cce0, no double quotes should be there instead be `version_bits=(${version//./ }"`.

I changed the workflow I use this action. Instead of letting it patch, I run it when I push tags (patch version myself), it then skips bumping the version but does everything else, so I did not realize this bug in new version. I will prepare an elegant fix as soon with proper testing.

Happy to know that someone else uses this action too, I thought I was the only one 😄 Thanks for the report!

**Comment by undergroundwires** [comment]
I just released [1.3.1](https://github.com/undergroundwires/bump-everywhere/releases/tag/1.3.1) for you. It should solve the problem. I added [a regression test](https://github.com/undergroundwires/bump-everywhere/blob/7066039799199d21b11d81de11e57a9494e83766/tests/bump-and-tag-version.test.sh#L15) to ensure it will keep working.

Feel free to verify it works now 😀

It's out-of-topic but I'd like to have better documentation to this repository and also understand how it's being used to maintain it. Could you explain to me how do you use the project? How does your automation work? I often use: Maintainer tags -> CI/CD trigger on tag push that runs bump-everywhere -> bump-everywhere does updates creates release -> CI/CD trigger on release creation that does other release actions needed (such as deployments).

**Comment by undergroundwires** [comment]
Closing this due to non-activity.
